### ISSUE TITLE
SIGTERM no propagated to process started with LOCALLY

### ISSUE THREAD

**Issue by gpoulin** [earthly/earthly#1684]
When sending a SIGTERM to earthly while running a `LOCALLY` target, earthly process will quit without sending the SIGTERM to the running local process 

For example, with the following `Earthfile`

```
VERSION 0.6

server:
    FROM python:3.9-slim
    ENTRYPOINT python -m http.server
    SAVE IMAGE server

serve:
    LOCALLY
    WITH DOCKER --load=+server
        RUN docker run --rm -p 127.0.0.1:8000:8000 server
    END
```

If running, `earthly +serve` and then `ctrl-c` after the server start, Earthly will quit with
`Error: Earthfile line 11:8 with docker run: unlazy force execution: Canceled: context canceled`. The started server will continue to run.

Ideally, when sending a SIGTERM to earthly while running a local process, it should forward it to the local process and wait for it to quit.

**Comment by vladaionescu** [comment]
Perhaps a second ctrl-c sends a SIGKILL.

**Comment by gpoulin** [comment]
Yeah, I think it makes sense. Having some clear message would help. Something like `Cancelling '+{task}'. Press 'ctrl-c' to force terminate the task`.

**Comment by immesys** [comment]
Is there any workaround for this? I was really hoping to not have to wrap earthly in a Makefile, but I have this:

```
# local-stack starts the compose stack for local development
local-stack:
    LOCALLY
    WAIT
        BUILD +img-app
        BUILD +img-web
        BUILD +img-postgres
    END
    RUN docker-compose -f test/stack/compose.yaml -p local-stack up
```

And if you ctrl-c out of it, it doesn't clean up the compose stack (unlike if you ctrl-c out of running the docker-compose command locally. So the only workaround I can think of is to wrap earthly in a Makefile and have just this one rule actually run from Make, not earthly

**Comment by alexcb** [comment]
> Is there any workaround for this?

unfortunately not; the best I can think of is something along the lines of

```
  RUN docker-compose -f test/stack/compose.yaml -p local-stack up & export dcpid=$! && echo "ctrl-c doesnt work, instead run \"kill $dcpid\" in a different terminal" && wait $dcpid

```

to make matters worse, `RUN --interactive` doesn't work under locally mode, which means you can't do something more clever with the sh script.

**Comment by venu** [comment]
https://github.com/earthly/earthly/issues/1684#issuecomment-1786806354

Is there workaround for this? or any ETA for fixing this? Thank you in advance!
